### PR TITLE
Feature/cluster datasource name

### PR DESCRIPTION
**Pull request by yannickstruyf3**
One of my clients requested the possibility to search clusters by name via the data sources. This code adds this functionality to the existing data_source_nutanix_cluster.go file.

Acceptance test included.

### CODE REVIEW

**Comment by yannickstruyf3** [comment]
@marinsalinas rebased master

File: nutanix/data_source_nutanix_cluster.go [hunk 1]
```diff
@@ -2,20 +2,21 @@ package nutanix
...
 
 func dataSourceNutanixCluster() *schema.Resource {
 	return &schema.Resource{
 		Read: dataSourceNutanixClusterRead,
 		Schema: map[string]*schema.Schema{
 			"cluster_id": {
-				Type:     schema.TypeString,
-				Required: true,
+				Type: schema.TypeString,
+				// Required: true,
```
Comment (marinsalinas) [inline nutanix/data_source_nutanix_cluster.go]: Could you remove this attribute and add Computed: true.

File: nutanix/data_source_nutanix_cluster.go [hunk 2]
```diff
@@ -99,8 +100,9 @@ func dataSourceNutanixCluster() *schema.Resource {
 				Computed: true,
 			},
 			"name": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:          schema.TypeString,
+				Optional:      true,
```
Comment (marinsalinas) [inline nutanix/data_source_nutanix_cluster.go:105]: Could you add `Computed:true` as well, and set the value on Read function.

File: nutanix/data_source_nutanix_cluster.go (hunk 1 above)
Comment (marinsalinas) [reply nutanix/data_source_nutanix_cluster.go]: also add `ConflictsWith: []string{"name"}` in order to raise an error when both values are set.

File: nutanix/data_source_nutanix_cluster.go [hunk 3]
```diff
@@ -1034,3 +1062,35 @@ func dataSourceNutanixClusterRead(d *schema.ResourceData, meta interface{}) erro
 
 	return nil
 }
+
+func findClusterByName(conn *v3.Client, name string) (*v3.ClusterIntentResponse, error) {
```
Comment (marinsalinas) [inline nutanix/data_source_nutanix_cluster.go:1054]: Nice catch, isn't there any way to find a cluster by name through the API?, If not we can keep like this and create a ticket to the API Tram to support this.

File: nutanix/data_source_nutanix_cluster.go [hunk 4]
```diff
@@ -589,15 +591,41 @@ func dataSourceNutanixClusterRead(d *schema.ResourceData, meta interface{}) erro
...
+	} else {
+		n, ok := d.GetOk("name")
+		if !ok {
+			return fmt.Errorf("please provide the cluster_id or name attribute")
+		}
+		v, err = findClusterByName(conn, n.(string))
+		if err != nil {
+			return err
+		}
+		// clusterEntitiesMetadata := &v3.DSMetadata{}
```
Comment (marinsalinas) [inline nutanix/data_source_nutanix_cluster.go]: please remove all the unnecessary comments.

**Review by marinsalinas — CHANGES_REQUESTED** [review]
Almost done, just few comments.

File: nutanix/resource_nutanix_virtual_machine.go [hunk 5]
```diff
@@ -1913,3 +1928,15 @@ func waitForIPRefreshFunc(client *v3.Client, vmUUID string) resource.StateRefres
 		return resp, WAITING, nil
 	}
 }
+
+func CountDiskListCdrom(dl []*v3.VMDisk) (int, error) {
+	log.Printf("=====")
```
Comment (marinsalinas) [inline nutanix/resource_nutanix_virtual_machine.go]: could you remove unnecessary logs?

File: nutanix/resource_nutanix_virtual_machine.go [hunk 6]
```diff
@@ -1913,3 +1928,15 @@ func waitForIPRefreshFunc(client *v3.Client, vmUUID string) resource.StateRefres
...
+	counter := 0
+	for _, v := range dl {
+		if *v.DeviceProperties.DeviceType == "CDROM" {
+			counter++
+		}
+	}
+	log.Printf("=====")
```
Comment (marinsalinas) [inline nutanix/resource_nutanix_virtual_machine.go]: same here

**Review by marinsalinas — CHANGES_REQUESTED** [review]
Just remove the unnecessary logs please.

**Review by marinsalinas — APPROVED** [review]

**Pull request merged**